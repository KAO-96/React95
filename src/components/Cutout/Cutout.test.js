import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import { assertPropTypes } from "check-prop-types";

import Cutout from "./Cutout";
import themes from "../common/themes";
import { insetShadow } from "../common";

describe("Component: Cutout", () => {
  const testProps = {
    className: "",
    shadow: false,
    children: "text",
    style: { color: "red" }
  };

  it("warns if prop are missing or the wrong type", () => {
    expect(() => { assertPropTypes(Cutout.propTypes, { ...testProps, className: 123 }, "prop", Cutout.className); }).toThrow()
    expect(() => { assertPropTypes(Cutout.propTypes, { ...testProps, shadow: "true" }, "prop", Cutout.shadow); }).toThrow()
    expect(() => { assertPropTypes(Cutout.propTypes, { ...testProps, children: () => null }, "prop", Cutout.children); }).toThrow()
    expect(() => { assertPropTypes(Cutout.propTypes, { ...testProps, style: "red" }, "prop", Cutout.style); }).toThrow()
  })

  it("renders", () => {
    const wrapper = shallow(<Cutout />);
    expect(wrapper.length).toBe(1);
  })

  it("renders children", () => {
    const wrapper = shallow(<Cutout>text</Cutout>);
    expect(wrapper.text()).toBe("text");
  })

  it("renders className", () => {
    const wrapper = shallow(<Cutout className="my-class" />);
    expect(wrapper.props().className).toBe("my-class");
  })

  it("renders style object", () => {
    const wrapper = shallow(<Cutout style={{ color: "red" }} />);
    expect(wrapper.props().style).toEqual({ color: "red" });
  })

  it("renders other props passed", () => {
    const wrapper = shallow(<Cutout a={1} b={2} c={3} />);
    expect(wrapper.props().a).toBe(1);
    expect(wrapper.props().b).toBe(2);
    expect(wrapper.props().c).toBe(3);
  })

  it("renders default styles", () => {
    const tree = renderer.create(<Cutout theme={themes.default} />).toJSON();
    expect(tree).toHaveStyleRule("border-left-color", themes.default.borderDark);
    expect(tree).toHaveStyleRule("border-top-color", themes.default.borderDark);
    expect(tree).toHaveStyleRule("border-right-color", themes.default.borderLightest);
    expect(tree).toHaveStyleRule("border-bottom-color", themes.default.borderLightest);
    expect(tree).toHaveStyleRule("border-left-color", themes.default.borderDarkest, { modifier: ":before" });
    expect(tree).toHaveStyleRule("border-top-color", themes.default.borderDarkest, { modifier: ":before" });
    expect(tree).toHaveStyleRule("border-right-color", themes.default.borderLight, { modifier: ":before" });
    expect(tree).toHaveStyleRule("border-bottom-color", themes.default.borderLight, { modifier: ":before" });
  })

  it("renders a matching snapshot", () => {
    const tree = renderer.create(<Cutout>text</Cutout>).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
