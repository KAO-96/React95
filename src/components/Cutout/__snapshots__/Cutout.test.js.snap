// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Cutout renders a matching snapshot 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
  padding: 2px;
  border-style: solid;
  border-width: 2px;
}

.c0:before {
  position: absolute;
  left: 0;
  top: 0;
  z-index: 1;
  content: "";
  width: calc(100% - 4px);
  height: calc(100% - 4px);
  border-style: solid;
  border-width: 2px;
  pointer-events: none;
  box-shadow: inset 3px 3px 10px rgba(0,0,0,0.2);
}

<div
  className="c0"
>
  text
</div>
`;
